java-checks/src/main/java/org/sonar/java/checks/PrivateFieldUsedLocallyCheck.java

```diff
@@ -0,0 +1,226 @@
+
+    return parent;
+  }
+
+  private void checkPrivateField(Symbol privateFieldSymbol) {
+    if (fieldsReadOnAnotherInstance.contains(privateFieldSymbol)) {
+      return;
+    }
+
+    for (Set<Symbol> fieldsLiveInMethodEntry : fieldsLiveInMethodEntryPerMethod.values()) {
+      if (fieldsLiveInMethodEntry.contains(privateFieldSymbol)) {
+        return;
+      }
+    }
+
+    reportIssue(((VariableTree) privateFieldSymbol.declaration()).simpleName(), String.format(MESSAGE, privateFieldSymbol.name()));
+  }
+
+  private void initMethodsLiveFields(TypeSymbol classSymbol) {
+    fieldsLiveInMethodEntryPerMethod = new HashMap<>();
+    fieldsReadOnAnotherInstance = new HashSet<>();
+
+    for (Symbol memberSymbol : classSymbol.memberSymbols()) {
+
+      if (memberSymbol.isMethodSymbol() && memberSymbol.declaration() != null) {
+        MethodTree methodTree = (MethodTree) memberSymbol.declaration();
+
+        if (methodTree.block() != null) {
+          CFG cfg = CFG.build(methodTree);
+          LiveFields liveFields = LiveFields.analyze(cfg);
+          fieldsLiveInMethodEntryPerMethod.put(methodTree, liveFields.getIn(cfg.entry()));
+          fieldsReadOnAnotherInstance.addAll(liveFields.fieldsReadOnAnotherInstance);
+        }
+      }
+    }
+  }
+
+  private static boolean isPrivateField(Symbol memberSymbol) {
+    return memberSymbol.isPrivate() && memberSymbol.isVariableSymbol();
+  }
+
+  private static class LiveFields extends LiveVariables {
+
+    private Set<Symbol> fieldsReadOnAnotherInstance = new HashSet<>();
+
+    private LiveFields(CFG cfg) {
+      super(cfg);
+    }
+
+    public static LiveFields analyze(CFG cfg) {
+      return getLiveVariables(new LiveFields(cfg));
+    }
+
+    @Override
+    protected void processBlock(Block block, Map<Block, Set<Symbol>> kill, Map<Block, Set<Symbol>> gen) {
+      fieldsReadOnAnotherInstance.addAll(fieldsReadOnAnotherInstance(block));
+      super.processBlock(block, kill, gen);
+    }
+
+    private static Set<Symbol> fieldsReadOnAnotherInstance(Block block) {
+      Set<Symbol> fieldsReadOnAnotherInstance = new HashSet<>();
+      for (Tree element : block.elements()) {
+        if (element.is(Kind.MEMBER_SELECT)) {
+          MemberSelectExpressionTree memberSelect = (MemberSelectExpressionTree) element;
+          if (memberSelect.expression().is(Kind.IDENTIFIER)) {
+            String objectName = ((IdentifierTree) memberSelect.expression()).name();
+            Symbol symbol = memberSelect.identifier().symbol();
+
+            if (!symbol.isStatic() && !"this".equals(objectName) && isField(symbol)) {
+              fieldsReadOnAnotherInstance.add(symbol);
+            }
+          }
+        }
+      }
+      return fieldsReadOnAnotherInstance;
+    }
+
+    @Override
+    protected void processBlockElement(Tree element, Set<Symbol> blockKill, Set<Symbol> blockGen, Set<Tree> assignmentLHS) {
+      Symbol symbol;
+      switch (element.kind()) {
+        case ASSIGNMENT:
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this switch case number of lines from 8 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1151)